src/ServerTelemetryChannel/ServerTelemetryChannel.cs

```diff
@@ -304,6 +304,8 @@ public void Initialize(TelemetryConfiguration configuration)
         {
             this.Transmitter.Initialize();
 
+            this.EndpointAddress = new Uri(configuration.Endpoint.Ingestion, "v2/track").AbsoluteUri;
```

Comment:
why construct a new Uri and back to string?  Config.EndPoint.Ingestion.AbsoluteUrl + "v2/track" would do. Validation for ingestion would have already happened .

Reply:
I'm relying on the Uri library to validate endpoints so I'm intentionally initializing a Uri first.